### PR TITLE
Remove Godot's boot splash on android

### PR DESCRIPTION
**Pull request by syntaxerror247**
After the update to the SplashScreen logic in this PR https://github.com/godotengine/godot/pull/92965, we now have two splash screens on Android. This dual splash screen setup does not provide a polished user experience.

This PR solves that problem by removing the second splash screen that is previously rendered by the engine. Now, only the Android system splash screen will be used, ensuring a consistent and streamlined startup.

<details><summary>Demo Video</summary>
<p>

https://github.com/user-attachments/assets/7121fd9f-d151-4d3e-b14b-fc87411fa77f

</p>
</details>

### CODE REVIEW

**Comment by syntaxerror247** [comment]
@AThousandShips I don't think bug is the right label. It would be more appropriate to label it as an `enhancement`.

**Comment by syntaxerror247** [comment]
Okay, but this issue was fully known at tha time and even discussed in that PR.

**Comment by AThousandShips** [comment]
Got it, you didn't say so so I had no idea

**Comment by dustdfg** [comment]
Tested on Android 10. And confirm that it removes editor splash. Though I am not sure that users need exactly it. As I heard OS splash can't be very customized (while godot splash is more customizable?). Anyway the PR itself works

**Comment by syntaxerror247** [comment]
@dustdfg Absolutely! While Godot's splash screen offers more customization options, having two splash screens feels like a significant drawback to me. The [Android splash screen](https://developer.android.com/develop/ui/views/launch/splash-screen) also provides some customization options. You can set your adaptive foreground and background in Godot export settings to update the icon and background color. Additionally, with Gradle build, you can incorporate custom animated icons and branding images.

**Comment by dustdfg** [comment]
I meant that maybe it is better to provider user a choice what splash screen to use?

**Comment by syntaxerror247** [comment]
They’ve done the same with this as well. Unfortunately, we don’t have a choice, and there’s no way to remove the Android splash screen.

**Comment by Alex2782** [comment]
Android API - Docs: https://developer.android.com/develop/ui/views/launch/splash-screen
I'm not yet sure what effects it has on **Android 12 and 13**, has anyone tested it yet?

--------------

```cpp
#if !defined(TOOLS_ENABLED) && (defined(WEB_ENABLED) || defined(ANDROID_ENABLED))
	bool show_logo = false;
```
PR https://github.com/godotengine/godot/pull/92965  -> Fixes https://github.com/godotengine/godot/issues/69317, https://github.com/godotengine/godot/issues/62863, https://github.com/godotengine/godot/issues/74348 
no more now?


I think it was also about the project settings (For example the “Duration” setting / "Display time")

image

**Comment by syntaxerror247** [comment]
@Alex2782 I have tested it with `Android 14` and its working as intended.

> #if !defined(TOOLS_ENABLED) && (defined(WEB_ENABLED) || defined(ANDROID_ENABLED))
> 	bool show_logo = false;
> PR https://github.com/godotengine/godot/pull/92965 -> Fixes https://github.com/godotengine/godot/issues/69317, https://github.com/godotengine/godot/issues/62863, https://github.com/godotengine/godot/issues/74348
> no more now?

AFAIK, this wasn't causing the black screen problem, and actual fix was to _install the new SplashScreen API_ which is still used.
Removing `defined(ANDROID_ENABLED)` was needed to the show logo on Godot's splash.

And Duration time is also not affected and will now work with the android splash (I have tested it).

**Comment by m4gr3d** [comment]
This is intended behavior. We want the Godot splash screen on Android to provide consistency between platforms. And given the restrictions imposed by Android, we cannot remove the default Android splash screen, as such the intended behavior is to have both splash screens.

If the developer does not want the Godot splash screen, they can disable it themselves from the project settings.

**Comment by m4gr3d** [comment]
TLDR; the Godot's splash screen takes priority. If you find a mean to disable the Android splash screen while remaining consistent with Android's guidelines, that may be an option (and PR) worth looking at.

**Comment by syntaxerror247** [comment]
@m4gr3d I agree that "Godot's splash screen takes priority". But having two splash screens seems like an issue.

> If the developer does not want the Godot splash screen, they can disable it themselves from the project settings.
How can i disable that? If I toggle off "show Image" it still shows a black screen(splash bg color) after `Android Splash` for some time.

**Comment by m4gr3d** [comment]
> How can i disable that? If I toggle off "show Image" it still shows a black screen(splash bg color) after Android Splash for some time.

Good point. 
I may be okay with keeping the second part of your PR which allows to keep the Android splash screen until the main loop has started. Though in order to make this customizable, the configuration would need to be being an export setting so it can be read in `onCreate` (the project settings are not available until after `onGodotSetupCompleted`).

> find a mean to disable the Android splash screen while remaining consistent with Android's guidelines

I'll also reach out to Google to see if this is possible. This should be the ideal way to go if this was possible.

**Comment by syntaxerror247** [comment]
> Though in order to make this customizable, the configuration would need to be being an export setting so it can be read in onCreate

Do you mean creating a toogle to enable or disable Godot splash?
